spec/requests/clients_spec.rb

```diff
@@ -218,6 +218,41 @@
       end
     end
 
+    context "transfer repository" do
+      let(:new_provider) { create(:provider, symbol: "QUECHUA", password_input: "12345") }
+      let!(:prefix)  { create(:prefix) }
+      let!(:client_prefix)  { create(:client_prefix, client: client, prefix: prefix) }
+      let!(:provider_prefix)  { create(:provider_prefix, provider: provider, prefix: prefix) }
+      let(:doi) { create_list(:doi,10, client: client) }
+
+
+      let(:params) do
+        { "data" => { "type" => "clients",
+                      "attributes" => {
+                        "mode" => "transfer",
+                        "targetId" => new_provider.symbol,
+                        "provider" => new_provider.symbol,
+                      } } }
+      end
+
+      it "updates the record" do
+        put "/clients/#{client.symbol}", params, headers
+
+        expect(last_response.status).to eq(200)
+        expect(json.dig("data", "attributes", "name")).to eq("My data center")
+        expect(json.dig("data", "relationships", "provider", "data", "id")).to eq("quechua")
+        expect(json.dig("data", "relationships", "prefixes", "data").first.dig("id")).to eq(prefix.uid)
+
+        get "/providers/#{provider.symbol}"
+
+        expect(json.dig("data", "relationships", "prefixes", "data")).to be_empty
```

Comment:
Again, add a check that the provider prefixes that are not transferred are still there.